onnxruntime/contrib_ops/cpu/skip_layer_norm.cc

```diff
@@ -45,18 +45,42 @@ Status SkipLayerNorm<T>::Compute(OpKernelContext* p_ctx) const {
 
   const auto& input_dims = input->Shape().GetDims();
   size_t input_dims_size = input_dims.size();
+  const auto& skip_dims = skip->Shape().GetDims();
+  size_t skip_dims_size = skip_dims.size();
+
+  if (skip_dims_size != 3 && skip_dims_size != 2) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                           "skip is expected to have 3 or 2 dimensions, got ", skip_dims_size);
+  }
+  
+  int skip_sequence_length = static_cast<int>(skip_dims[skip_dims_size - 2]);
+
+  if (input->Shape() != skip->Shape() && skip_dims[0] != 1 && skip_dims[0] != skip_sequence_length) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                           "skip is expected to have same shape as input or a batch size of 1 or no batch size");
+  }
+
   if (input_dims_size != 3 && input_dims_size != 2) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                            "input is expected to have 3 or 2 dimensions, got ", input_dims_size);
   }
 
+
+  if (skip_sequence_length != input_dims[1] && skip_dims[skip_dims_size - 2] != 1) {
```

Comment:
skip_seq_len doesn't need to be 1, it's the sequence length:
the added skip shape: (1, sequence_length, hidden_size), (sequence_length, hidden_size)